### PR TITLE
FIX: Uniform array support

### LINKED ISSUE #345: Uniform arrays not supported in gloo
Here is a minimal example:

```python
import numpy as np
from vispy import app
from vispy import gloo

vertex = """
uniform float u_arr[2];
void main() {
    gl_Position = vec4(0, 0, 0, 1);
}
"""

voronoi_frag = """
void main() {
    gl_FragColor = vec4(1., 1., 1., 1.0);
}
"""

class Canvas(app.Canvas):
    def __init__(self):
        app.Canvas.__init__(self, close_keys='escape')
        self.program = gloo.Program(vertex, voronoi_frag)
        self.program['u_arr'] = (1., 1.)

    def on_initialize(self, event):
        gloo.clear('white')

if __name__ == "__main__":
    c = Canvas()
    c.show()
    app.run()
```

ping @rougier @almarklein

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #345.

### CODE REVIEW

File: vispy/gloo/program.py [hunk 1]
```diff
@@ -299,6 +300,20 @@ def __setitem__(self, name, data):
                     # Store and send GLIR command
                     self._user_variables[name] = data
                     self._glir.command('UNIFORM', self._id, name, type, data)
+ 
+            elif kind == 'uniform_array':
+                # Normal uniform; convert to np array and check size
+                dtype, numel = self._gtypes[type]
+                data = np.atleast_2d(data).astype(dtype)
+                print(data)
```
Comment (almarklein) [inline vispy/gloo/program.py]: remove print

**Comment by almarklein** [comment]
@rossant I'll let you have a look and do the honor of pushing that button.

**Comment by almarklein** [comment]
LGTM

**Comment by Eric89GXL** [comment]
@rossant this one is good to go, too, and Travis is happy.

**Comment by rossant** [comment]
Actually it might be better not to merge this before the big collections PR, otherwise we'll have to rebase it again...

**Comment by Eric89GXL** [comment]
Ahh, good point. I'll wait then.

**Pull request merged**